generic_config_updater/services_validator.py

```diff
@@ -17,13 +17,42 @@ def set_verbose(verbose=False):
 
 def _service_restart(svc_name):
     rc = os.system(f"systemctl restart {svc_name}")
-    logger.log(logger.LOG_PRIORITY_NOTICE,
-            f"Restarted {svc_name}", print_to_console)
+    if rc != 0:
+        # This failure is likely due to too many restarts
+        #
+        rc = os.system(f"systemctl reset-failed {svc_name}")
+        logger.log(logger.LOG_PRIORITY_ERROR, 
+                f"Service has been reset. rc={rc}; Try restart again...",
+                print_to_console)
+
+        rc = os.system(f"systemctl restart {svc_name}")
+        if rc != 0:
+            # Even with reset-failed, restart fails.
+            # Give a pause before retry.
+            #
+            logger.log(logger.LOG_PRIORITY_ERROR,
+                    f"Restart failed for {svc_name} rc={rc} after reset; Pause for 10s & retry",
+                    print_to_console)
+            os.system("sleep 10s")
+            rc = os.system(f"systemctl restart {svc_name}")
+
+    if rc == 0:
+        logger.log(logger.LOG_PRIORITY_NOTICE,
+                f"Restart succeeded for {svc_name}",
+                print_to_console)
+    else:
+        logger.log(logger.LOG_PRIORITY_ERROR,
+                f"Restart failed for {svc_name} rc={rc}",
+                print_to_console)
     return rc == 0
 
 
 def rsyslog_validator(old_config, upd_config, keys):
-    return _service_restart("rsyslog-config")
```

Comment:
>_service_restart [](http://example.com/codeflow?start=11&length=16)

With `_service_restart` fixed, do we still need this fix?

Reply:
What we need is to run /usr/bin/rsyslog-config.sh. 
What it does is, update /etc/rsyslog.conf from CONFG-DB & restart rsyslog.
We do the same and additionally handle the possibility of rsyslog restart failure.

rsyslog-config is not a service, but a one shot wrapper to run /usr/bin/rsyslog-config.sh, after updategraph.service, at the startup.